### PR TITLE
Update the readme in core for the UI and update the UI readme

### PR DESCRIPTION
**Pull request by richrace**
- Updated README in
  - UI
  - e2e tests
  - Top level core
  - API
- Added README in
  - Common
- Made the UI Makefile work again

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -4,20 +4,37 @@ The code to replace the processing logic of Bichard 7.
...
-- [Testing](#testing)
-- [Excluding Triggers](#excluding-triggers)
-- [Comparing New and Old Bichard](#comparing-new-and-old-bichard)
-  - [Checking a comparison file](#checking-a-comparison-file)
-  - [Checking a comparison file on old Bichard](#checking-a-comparison-file-on-old-bichard)
-  - [Comparing outputs locally](#comparing-outputs-locally)
-  - [Configuration](#configuration)
-- [Conductor](#conductor)
+- [Bichard 7 Core](#bichard-7-core)
+  - [Contents](#contents)
```
Comment (chubberlisk) [inline README.md:8]: I don't think these links are useful and can be removed.

File: README.md [hunk 2]
```diff
@@ -94,7 +111,17 @@ We can't pull the images down from ECR for an M1 Mac because they are not in ARM
 1. In the [bichard7-next-infrastructure-docker-images](https://github.com/ministryofjustice/bichard7-next-infrastructure-docker-images/) repository, run `make build-local` to just build the required images
 1. Follow the instructions in the [bichard7-next](https://github.com/ministryofjustice/bichard7-next/#building-liberty-on-arm) repository to build the Bichard Open Liberty image
 1. In the [bichard7-next-user-service](https://github.com/ministryofjustice/bichard7-next-user-service/) repository run `make build`
-1. In the [bichard7-next-ui](https://github.com/ministryofjustice/bichard7-next-ui/) repository, run `make build`
+1. In the [bichard7-next-ui](https://github.com/ministryofjustice/bichard7-next-core/tree/main/packages/ui) package, run `make build` or the top level of [bichard7-next-core](https://github.com/ministryofjustice/bichard7-next-core) run this script `./scripts/build-ui-docker.sh`
+
+## Running Packages locally
+
+1. From the root directory `npm ci`
```
Comment (chubberlisk) [inline README.md:118]: Not sure if it's worth mentioning but I always use `npm i` and never had any problems.

File: README.md [hunk 3]
```diff
@@ -94,7 +111,17 @@ We can't pull the images down from ECR for an M1 Mac because they are not in ARM
...
+2. Run `npm run build:core`
+3. If you need to change `packages/common` you have to two options, `cd packages/common`
+   1. And run `npm run build`
+   2. Or `npm run watch`
```
Comment (chubberlisk) [inline README.md:122]: Are we able to do `npm run build -w packages/common` and `npm run watch -w packages/common`? Or there an annoying thing where it's expecting to be in a certain directory?

File: packages/common/README.md [hunk 4]
```diff
@@ -0,0 +1,17 @@
+# @moj-bichard7/common
+
+- [@moj-bichard7/common](#moj-bichard7common)
+  - [What is the package?](#what-is-the-package)
+  - [Development](#development)
+    - [When do I need to build this?](#when-do-i-need-to-build-this)
+
```
Comment (chubberlisk) [inline packages/common/README.md:7]:
```suggestion

## Contents

- [What is the package?](#what-is-the-package)
- [Development](#development)
  - [When do I need to build this?](#when-do-i-need-to-build-this)

```

File: packages/e2e-test/README.md [hunk 5]
```diff
@@ -1,19 +1,22 @@
...
-  - [Against E2E Test environment](#against-e2e-test-environment)
-  - [Against pre-production](#against-pre-production)
-  - [Configuration parameters](#configuration-parameters)
-  - [Debugging](#debugging)
-  - [VS Code Extension](#vs-code-extension)
-- [Running the characterisation tests](#running-the-characterisation-tests)
-  - [For Core](#for-core)
-  - [For legacy Bichard](#for-legacy-bichard)
+- [@moj-bichard7/e2e-tests](#moj-bichard7e2e-tests)
+  - [Contents](#contents)
```
Comment (chubberlisk) [inline packages/e2e-test/README.md:8]: Same here, I don't think these are useful.

File: packages/e2e-test/README.md [hunk 6]
```diff
@@ -26,19 +29,37 @@ Before running the tests locally, you need to make sure that the environment is
...
 Run `npm install` to download the correct Chromium revision (856583), then run `node node_modules/puppeteer/install.js`.
 
+#### Cucumber test individual features
+
+To run a Cucumber test locally we need to have all docker images built and running. We need any of the latest changes.
+
+For example, if we want to test a change in the Next UI, we need to kill the current docker container for the UI. Build
+a new docker image that contains the new changes. And then boot it up from the root of the core directory using `npm run all`.
+
+We can then run this command:
```
Comment (chubberlisk) [inline packages/e2e-test/README.md:52]: In general, if there are steps then we should use numbered bullet points to make it easier to follow and read.

```suggestion
To run a Cucumber test locally you need to have all Docker images built and running with the latest changes.

For example, if you want to test a change in the Next UI:

1. Kill the current Docker container for the UI.
1. [Build a new Docker image that contains the new changes](https://github.com/ministryofjustice/bichard7-next-core/blob/main/packages/ui/README.md#building-the-docker-image). 1. Run `npm run all` at the root of the Core repository to boot everything up again.

You can then run this command:
```

File: packages/ui/README.md [hunk 7]
```diff
@@ -1,7 +1,24 @@
-# bichard7-next-ui
+# @moj-bichard7/ui
 
 This is a [Next.js](https://nextjs.org/) project bootstrapped with [`create-next-app`](https://github.com/vercel/next.js/tree/canary/packages/create-next-app).
 
+## Contents
+
+- [@moj-bichard7/ui](#moj-bichard7ui)
+  - [Contents](#contents)
```
Comment (chubberlisk) [inline packages/ui/README.md:8]: Same here!

File: packages/ui/README.md [hunk 8]
```diff
@@ -37,6 +54,9 @@ Once the dependencies have run follow these login instruction:
 
 ## Development
 
+You need to have built `bichard7-next-core/packages/core` and `bichard7-next-core/packages/common` before you can run
+this package.
```
Comment (chubberlisk) [inline packages/ui/README.md:58]: Maybe link to your new "Running Packages locally" section in the root README?

File: packages/ui/README.md [hunk 9]
```diff
@@ -63,6 +83,16 @@ To run code-based (non-visual, no components get rendered) unit tests, run
     npm run test:unit
 ```
 
+#### React Component Testing
+
+You should have the UI running locally (`npm run dev`) and run the following:
+
+```bash
+    npm run cypress:open
```
Comment (chubberlisk) [inline packages/ui/README.md:91]: Couple of other commands in this README with unnecessary indentation... 👀 

```suggestion
npm run cypress:open
```

File: packages/ui/README.md [hunk 10]
```diff
@@ -83,8 +113,14 @@ To run the `cypress` tests against this, run:
...
 ```
 
-or if you're running the UI locally (`npm run dev` - will require you to kill the UI docker container first):
+or if you're running the UI locally (`npm run dev`):
 
 ```bash
     npm run cypress:open
 ```
+
+## Building the docker image
```
Comment (chubberlisk) [inline packages/ui/README.md:122]: Sorry... 😅 

```suggestion
## Building the Docker image
```

File: packages/ui/README.md [hunk 11]
```diff
@@ -83,8 +113,14 @@ To run the `cypress` tests against this, run:
...
+
+From the root level of core: `./scripts/build-ui-docker.sh`
```
Comment (chubberlisk) [inline packages/ui/README.md:124]:
```suggestion
From the root level of Core: `./scripts/build-ui-docker.sh`
```

File: README.md (hunk 1 above)
Comment (richrace) [reply README.md:8]: I like having a Table of Contents 🤷‍♂️

File: README.md (hunk 2 above)
Comment (richrace) [reply README.md:118]: How about `npm i --workspaces --include-workspace-root`?

File: packages/common/README.md (hunk 4 above)
Comment (richrace) [reply packages/common/README.md:7]: I have an auto-generate table of contents feature in my editor. I can remove that top level if you like.

File: packages/ui/README.md (hunk 8 above)
Comment (richrace) [reply packages/ui/README.md:58]: I think I'll link to this.

File: README.md (hunk 1 above)
Comment (chubberlisk) [reply README.md:8]: Not saying I don't like a table of contents, just those specific two links. (I was the one who added a TOC in the first place! 😝)

File: packages/common/README.md (hunk 4 above)
Comment (chubberlisk) [reply packages/common/README.md:7]: Yes please, I think the top level one is just noise and adds unnecessary extra indentation.

File: README.md (hunk 2 above)
Comment (chubberlisk) [reply README.md:116]: Seems fine? I deleted my `node_modules` and then ran `npm i --workspaces --include-workspace-root`. I was able to run `npm run test:unit -w packages/core`. Not sure the best way to test...

File: README.md (hunk 3 above)
Comment (chubberlisk) [reply README.md:120]: Okay cool, maybe include both?

**Pull request merged**